nixos/modules/security/pam.nix

```diff
@@ -1546,9 +1549,9 @@ in
         (name: "r ${config.environment.etc."pam.d/${name}".source},\n")
         (attrNames config.security.pam.services) +
       ''
-      mr ${getLib pkgs.pam}/lib/security/pam_filter/*,
-      mr ${getLib pkgs.pam}/lib/security/pam_*.so,
-      r ${getLib pkgs.pam}/lib/security/,
+      mr ${getLib package}/lib/security/pam_filter/*,
+      mr ${getLib package}/lib/security/pam_*.so,
+      r ${getLib package}/lib/security/,
```

Comment:
I think your change makes these rules unnecessary. The section below adds apparmor rules for any configured modules with absolute paths, which is now all of them. So, I think we could remove these modified rules, and then convert the absolute path filter below into an assertion requiring all `modulePath` to be absolute.

Reply:
i've deleted this code and added a **warning** to the below codepath whenever we encounter non-absolute paths.

i can do an assertion right here if you're confident, but since i'm not versed in apparmor setups my inclination is to instead show a warning, and  open a "(NOT FOR MERGE)" PR immediately after this one to upgrade that to an assertion. this allows users at least a couple weeks to fix any broken configs/provide feedback before we fail eval for anyone.